components/properties/src/provider.rs

```diff
@@ -311,11 +317,140 @@ impl<'data, T: TrieValue> PropertyCodePointMapV1<'data, T> {
     }
 }
 
+/// This is a property name that can be "loose matched" as according to
+/// https://www.unicode.org/Public/UCD/latest/ucd/PropertyValueAliases.txt
+///
+/// (matched case-insensitively in ASCII, ignoring underscores, whitespace, and hyphens)
+///
+/// This is expected to be ASCII, but we do not rely on this invariant anywhere except during
+/// datagen.
+#[derive(PartialEq, Eq)] // VarULE wants these to be byte equality
+#[derive(VarULE)]
+#[repr(transparent)]
+pub struct NormalizedPropertyNameStr([u8]);
+
+#[cfg(feature = "datagen")]
+impl serde::Serialize for NormalizedPropertyNameStr {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        use serde::ser::Error;
+        if serializer.is_human_readable() {
+            let s = str::from_utf8(&self.0)
+                .map_err(|_| S::Error::custom("Attempted to datagen invalid string property"))?;
+            serializer.serialize_str(s)
+        } else {
+            serializer.serialize_bytes(&self.0)
+        }
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> serde::Deserialize<'de> for Box<NormalizedPropertyNameStr> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        use alloc::borrow::Cow;
+        let s; // lifetime lengthening
+        let b;
+        // Can be improved with https://github.com/unicode-org/icu4x/issues/2310
+        // the allocations here are fine, in normal ICU4X code they'll only get hit
+        // during human-readable deserialization
+        let bytes = if deserializer.is_human_readable() {
+            s = <Cow<str>>::deserialize(deserializer)?;
+            s.as_bytes()
+        } else {
+            b = <Cow<[u8]>>::deserialize(deserializer)?;
```

Comment:
Observation: I think this code path shouldn't actually run because these things are in a VarZeroVec so machine-readable should not even touch this Deserialize impl

Reply:
Yes, it shouldn't, but I have it in for completeness.